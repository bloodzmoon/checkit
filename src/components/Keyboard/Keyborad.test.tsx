import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import { IKeycap } from 'keycap'
import { Keyboard, Keycap } from '@components'

describe('<Keyboard>', () => {
  const mock: IKeycap[] = [{ text: 'A', size: '1' }]

  it('should render layout correctly', () => {
    const wrapper = shallow(<Keyboard layout={mock} />)
    const child = wrapper.children()
    expect(child.type()).toBe(Keycap)
    expect(child.length).toBe(1)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Keyboard layout={mock} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
