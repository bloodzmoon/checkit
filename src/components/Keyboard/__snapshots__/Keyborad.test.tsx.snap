// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Keyboard> should match snapshot 1`] = `
<div
  className="layout keyboard"
>
  <span
    className="base size1"
  >
    A
  </span>
</div>
`;
